.github/workflows/esy-build.yml

```diff
@@ -3,28 +3,36 @@ name: Build using esy
 on:
   pull_request:
   push:
-   branches:
-     - fork
+    branches:
+      - fork
 
 jobs:
   build:
     name: Build
 
     strategy:
       matrix:
-        system: [windows-latest, ubuntu-latest, macos-latest]
+        system: [windows, ubuntu, macos, macos-arm64]
 
-    runs-on: ${{ matrix.system }}
+    runs-on: ${{ matrix.system }}-latest
 
     steps:
       - uses: actions/checkout@v2
       - uses: actions/setup-node@v2
+        if: ${{ matrix.system != 'macos-arm64' }}
         with:
           node-version: 14
 
       - name: Install esy
+        if: ${{ matrix.system != 'macos-arm64' }}
         run: npm install -g esy
 
       - uses: esy/github-action@master
+        if: ${{ matrix.system != 'macos-arm64' }}
+        with:
+          cache-key: ${{ hashFiles('esy.lock/index.json') }}
+
+      - uses: EduardoRFS/github-action@0e5336e318d2f648cf881ab85ece10f51e369b17
```

Comment:
what does this change? are you gonna submit to master?

Reply:
Because of the weird transition from arm64 to x86_64 when using rosetta I need to explicitly call "arch -arm64 esy" this is what this branch does